### ISSUE TITLE
Cannot load SFrame from s3 with a space in the filename

### ISSUE THREAD

**Issue by srikris** [apple/turicreate#1660]
```
import turicreate
data = turicreate.SFrame('s3://test/file with space.sframe')
```
results in a file not found error.

**Comment by Jarvi-Izana** [comment]
The s3 url is parsed correctly.
`bucket = "test", object_name = "file with space.sframe"`. 
It's using '/' to [parse tokens](https://github.com/apple/turicreate/blob/master/src/core/storage/fileio/s3_api.cpp#L162
).  I think the whitespace won't mess up the parsing logic to get the right key. 

What happens is that the file is not found in s3 after a list operation applied with the key.
[list_objects_response response = list_objects(url, proxy);](https://github.com/apple/turicreate/blob/master/src/core/storage/fileio/s3_api.cpp#L574)

**Comment by TobyRoseman** [comment]
This is now working in master. It was likely fixed in our 6.2 release.
